### DISCUSSION TITLE
Github Pages hasn't recognized "***.io.vn" is APEX domain yet

### DISCUSSION THREAD

**Discussion by timhangnhanh** [community/community#57286] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Github Pages hasn't recognized "***.io.vn" is APEX domain yet so I cannot setup Custom Domain using my new domain abc.io.vn that I just bought. Please help me to update it.

**Comment by yoannchaudet** [comment]
Do you mind opening a support request?

This is how we do the validation:

https://github.com/github/pages-health-check/blob/c988b05f184aac65960cbb5d191cfa7134a947aa/lib/github-pages-health-check/domain.rb#L187-L189

And unless I am missing something, `io.vn` is a valid public suffix:

<img width="561" alt="image" src="https://github.com/community/community/assets/14911070/8de72bda-197c-43e5-8926-52a9d9490360">

**Reply by Doorem** [reply]
No

**Comment by timhangnhanh** [comment]
Hi @yoannchaudet ,

Could you give me the suggestion? It looks like that Github Pages has still determined `io.vn` is a domain, not a public suffix yet.
Thank you so much!

![image](https://github.com/community/community/assets/133499788/c088b082-0f50-4f67-b692-6076f106b6ea)

![image](https://github.com/community/community/assets/133499788/7d10f01b-1922-42f1-9188-d35084313679)

**Reply by yoannchaudet** [reply]
On my side, `timhangnhanh.io.vn` does not resolve to anything.

<img width="1140" alt="image" src="https://github.com/community/community/assets/14911070/f398ddaa-efea-4ecc-bbbd-482618819692">

Are you sure this is the domain you own? Did you create any records on it yet?

**Reply by timhangnhanh** [reply]
Thank @yoannchaudet,

There are some issues from my side, maybes they are reasons why you could not resolve timhangnhanh.io.vn.
I have fixed it and current result is better, domain is verified but DNS check process is still unsuccessful.
Could you take a look at them and give me the suggestion, thank you so much.

<img width="284" alt="image" src="https://github.com/community/community/assets/133499788/4469ac1a-8e35-445d-b840-97e94e6ee6f2">

![image](https://github.com/community/community/assets/133499788/c0e06c02-164c-4598-8a25-080239905c26)

<img width="786" alt="image" src="https://github.com/community/community/assets/133499788/56c82a87-3b51-4204-bc18-71075d9d4b4c">

**Reply by airtower-luna** [reply]
You need to point the A and AAAA records for your domain at the Pages servers (IPv4 look right, [IPv6 is missing some](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain)), not just its `www` subdomain. If you want the subdomain to work, too, [you need to set up a CNAME record](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain-and-the-www-subdomain-variant).

**Comment by timhangnhanh** [comment]
Thank @airtower-luna, @yoannchaudet,

My site is live, but HTTPS still waited for processing in 24 hours.
My last sites didn't need too much time for this process. I remember that it just took several minutes.

**Reply by yoannchaudet** [reply]
Please change `www` to point to `thn-timhangnhanh.github.io` as per https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain-and-the-www-subdomain-variant.

**Reply by timhangnhanh** [reply]
Thank @yoannchaudet,

I have corrected this configuration,
but enforcing HTTPS is still not available for my site.
This process have taken more than 24 hours,
I don't know what to do next. Could you help me!!!

**Reply by yoannchaudet** [reply]
Your setup is now valid. I did look at telemetry and the `io.vn` domain is currently being rate limited by Let's Encrypt. More on that in https://community.letsencrypt.org/t/register-cert-for-new-domain/200223.

It was only added to the public suffix list a few days ago and I assume Let's Encrypt has not picked up this change yet to adjust the rate limit.

We do try to request a cert on your behalf each time it is needed when you display the Pages settings on your repo. You can go to that page in the new few days a few times.

**Reply by timhangnhanh** [reply]
Thank you so much @yoannchaudet,

Let me do it sometime in next days.

**Comment by kinhchendev** [comment]
Hi @timhangnhanh , I faced the same issue with you. I registered domain kinhdev.io.vn and after two days, it still showing the DNS Checking unsuccessful although I can access my website via kinhdev.io.vn with https warning. Did you find the solution for this issue?
